src/main/java/graphql/execution/instrumentation/dataloader/FieldLevelTrackingApproach.java

```diff
@@ -32,63 +29,60 @@
 
     private static class CallStack implements InstrumentationState {
 
-        private final Map<Integer, Integer> expectedFetchCountPerLevel = new LinkedHashMap<>();
-        private final Map<Integer, Integer> fetchCountPerLevel = new LinkedHashMap<>();
-        private final Map<Integer, Integer> expectedStrategyCallsPerLevel = new LinkedHashMap<>();
-        private final Map<Integer, Integer> happenedStrategyCallsPerLevel = new LinkedHashMap<>();
-        private final Map<Integer, Integer> happenedOnFieldValueCallsPerLevel = new LinkedHashMap<>();
-
+        private final IntMap expectedFetchCountPerLevel = new IntMap();
+        private final IntMap fetchCountPerLevel = new IntMap();
+        private final IntMap expectedStrategyCallsPerLevel = new IntMap();
+        private final IntMap happenedStrategyCallsPerLevel = new IntMap();
+        private final IntMap happenedOnFieldValueCallsPerLevel = new IntMap();
 
         private final Set<Integer> dispatchedLevels = new LinkedHashSet<>();
 
         CallStack() {
-            expectedStrategyCallsPerLevel.put(1, 1);
+            expectedStrategyCallsPerLevel.increment(1, 1);
```

Comment:
I think you should make another method that is a "set".  The above code is "set level 1 to the value 1" whereas your code is "increment" the value by 1 (but its zero right now)

So I think a `expectedStrategyCallsPerLevel.set(1)` is a better method